### ISSUE TITLE
support JP2 master files for ingest

### ISSUE THREAD

**Issue by ebenenglish** [marriott-library/newspaper_works#41]
(This was split off from #28.)

**Descriptive summary**
Users should be able to upload/ingest newspaper content when the master file is JP2. The upload should create the remaining derivatives (PDF, thumbnail, ALTO, etc.)

**Background:**

From #39:
Hyrax does not easily support JP2 master/source files OOTB on Ubuntu, since it uses ImageMagick exclusively for hydra-derivatives, and ImageMagick on Ubuntu does not support JP2 decoding. _JP2 file upload breaks thumbnail creation OOTB on Ubuntu_. This may be a reason to (eventually, sooner or later) make our own thumbnail maker to replace what Hyrax does OOTB (one that falls back to graphicsmagick, so we can support JP2 masters without telling consumers of newspaper_works to build their own ImageMagick from source).

From [Slack](https://mlib-its.slack.com/archives/C7S7RL7NV/p1532534165000623):
```
Issue here is that the OOTB thumbnail generation will choke on JP2-as-master.  
hydra-derivatives uses MiniMagick, but does not support its graphicsmagick backend, 
so the JP2-master use case is limited OOTB by operating environment...

...unless we choose to override OOTB derivative creation for thumbnails, which may 
not be a big investment of time.
(that time investment is not in writing a thumbnail making service, but in blocking the 
default service from running, just for one work type)
```

**Expected behavior**
As an admin user
When I am on the add/edit form for an object
And I add an item with a JP2 as the source file
And click Save
The system should ingest the uploaded file
And create all necessary derivatives (TIF, PDF, ALTO, Annotation List)

**Comment by ebenenglish** [comment]
Closely related to #97.

**Comment by ebenenglish** [comment]
Fixed via #186.

**Comment by ebenenglish** [comment]
Although this ticket is closed, it is important to note that JP2 files are only supported for batch ingest, and not supported for single item uploads via the browser/UI.

TODO: create another ticket to support JP2 upload via browser/UI, in which:
 - TIFF master files are not created as part of the process (JP2 remains the true 'master')
 - thumbnails are created directly from the JP2 files

With regard to the latter, the primary issue is that Hyrax OOTB derivative plugin (we use only for thumbnails, but apps might use it for more) uses ImageMagick (via `hydra-derivatives` -> `minimagick`), which won't decode JP2 at all on either Ubuntu 16.04 or 18.04.

One approach would be to make a thumbnail plugin that uses `opj_decompress` to intermediate/tmp TIFF, plus imagemagick to thubnail jpg; toggle off the default Hyrax `FileSetDerivativeService` unless app consuming NewspaperWorks decides in its config to turn it back on).
